src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -1162,6 +1165,28 @@ object HashMap extends ImmutableMapFactory[HashMap] with BitOperations.Int {
         super.++=(xs)
     }
 
+    def getOrElse[V1 >: B](a: A, f: => V1): V1 = getOrElse0(rootNode, a, computeHash(a), 0, f)
+    def result[V1](f: B => V1): HashMap[A, V1] = {
+      val workList = mutable.ArrayStack[HashMap[A, B]]()
```

Comment:
we are going to create lots of worklists. Can we just create one in the caller and reuse
Ideally with a Array based srtructure so push doesnt create an object